Sources/TootSDK/TootClient/TootClient.swift

```diff
@@ -458,11 +463,38 @@ extension TootClient {
 extension TootClient {
     /// Uses the currently available credentials to connect to an instance and detect the most compatible server flavour.
     public func connect() async throws {
-        if let flavour = await flavourFromNodeInfo() {
-            self.flavour = flavour
+        if let nodeInfo = await getNodeInfoIfAvailable() {
```

Comment:
This check looks problematic to me. The idea was that we prefer to use node info for flavour detection since it seemed to be the best place for that. Now that we also want to access apiVersions from instance v2 I think we should prefer that approach instead. Node info could be a fallback.